packages/client/hmi-client/src/services/models/simulation-service.ts

```diff
@@ -171,20 +173,31 @@ export async function getRunResultCiemss(runId: string, filename = 'result.csv')
 	parsedRawData.forEach((inputRow) => {
 		const outputRowRunResults = { timestamp: inputRow.timepoint_id };
 		Object.keys(inputRow).forEach((key) => {
-			const keyArr = key.split('_');
-			const keySuffix = keyArr.pop();
-			const keyName = keyArr.join('_');
-
-			if (keySuffix === 'param' || keySuffix === 'state') {
-				outputRowRunResults[keyName] = inputRow[key];
-				if (!runConfigs[keyName]) {
-					runConfigs[keyName] = [];
+			if (key.endsWith('_observable_state')) {
+				const newKey = key.replace(/_observable_state$/, '');
+				outputRowRunResults[newKey] = inputRow[key];
+				if (!runConfigs[newKey]) {
+					runConfigs[newKey] = [];
 				}
-				runConfigs[keyName].push(Number(inputRow[key]));
-			} else if (keySuffix === 'sol') {
-				outputRowRunResults[keyName] = inputRow[key];
-			} else if (keySuffix === 'obs') {
-				outputRowRunResults[keyName] = inputRow[key];
+				runConfigs[newKey].push(Number(inputRow[key]));
+				return;
+			}
+			if (key.endsWith('_state')) {
+				const newKey = key.replace(/_state$/, '');
+				outputRowRunResults[newKey] = inputRow[key];
+				if (!runConfigs[newKey]) {
+					runConfigs[newKey] = [];
+				}
```

Comment:
Should this test be done before those `if`s?

Reply:
nope, each if block generates a different 'newKey'